### PR TITLE
Add tests for shared attribute classes

### LINKED ISSUE #50: Add shared misc classes and skeleton structure for refactor

### PR DESCRIPTION
**Pull request by RuiFengg**
Tests added for Amount, Date, Description constructor and validity checks

### CODE REVIEW

File: src/test/java/seedu/address/model/util/attributes/AmountTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Test
+    public void constructor_invalidAmount_throwsIllegalArgumentException() {
+        String invalidAmount = "";
+        assertThrows(IllegalArgumentException.class, () -> new Amount(invalidAmount));
+    }
+
+    @Test
+    public void isValidAmount() {
```
Comment (hanstirtaputra) [inline src/test/java/seedu/address/model/util/attributes/AmountTest.java:22]: Should we follow the three part format for naming of test methods?

File: src/test/java/seedu/address/model/util/attributes/DateTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Test
+    public void constructor_invalidDate_throwsIllegalArgumentException() {
+        String invalidDate = "";
+        assertThrows(IllegalArgumentException.class, () -> new Date(invalidDate));
+    }
+
+    @Test
+    public void isValidDate() {
```
Comment (hanstirtaputra) [inline src/test/java/seedu/address/model/util/attributes/DateTest.java:23]: Likewise for the naming of the test method

File: src/test/java/seedu/address/model/util/attributes/DateTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        // valid dates
+        assertTrue(Date.isValidDate("20-1-2020"));
+        assertTrue(Date.isValidDate("20-12-2020"));
+        assertTrue(Date.isValidDate("2-1-2020"));
+        assertTrue(Date.isValidDate("2-12-2020"));
+
+    }
+
+    @Test
+    public void toStringTest() {
```
Comment (hanstirtaputra) [inline src/test/java/seedu/address/model/util/attributes/DateTest.java:44]: Similarly for the naming of test method

File: src/test/java/seedu/address/model/util/attributes/DescriptionTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @Test
+    public void constructor_invalidDescription_throwsIllegalArgumentException() {
+        String invalidDescription = "";
+        assertThrows(IllegalArgumentException.class, () -> new Description(invalidDescription));
+    }
+
+    @Test
+    public void isValidDescription() {
```
Comment (hanstirtaputra) [inline src/test/java/seedu/address/model/util/attributes/DescriptionTest.java:22]: Same here for the naming of test method

**Review by hanstirtaputra — CHANGES_REQUESTED** [review]
Methods look good. Should we follow the three part naming format of test methods for all the test methods?

File: src/test/java/seedu/address/model/util/attributes/AmountTest.java (hunk 1 above)
Comment (RuiFengg) [reply src/test/java/seedu/address/model/util/attributes/AmountTest.java:22]: Yup its quite referenced from the original AB3 tests

**Review by yanlynnnnn — APPROVED** [review]
LGTM

**Pull request merged**